init.lua

```diff
@@ -30,6 +30,20 @@ local function destroy(handle)
 end
 ffi.gc(cudnn.handle, destroy)
 
+cudnn.typemap = {
+   ['torch.CudaHalfTensor']   = 'CUDNN_DATA_HALF',
+   ['torch.CudaTensor']       = 'CUDNN_DATA_FLOAT',
+   ['torch.CudaDoubleTensor'] = 'CUDNN_DATA_DOUBLE',
+}
+
+-- TODO: determine if device supports true half and use true half on it
```

Comment:
I have merged your changes into my clones. WIll test on 6.0 tomorrow. 
Regarding cunn:
- We shoud submit a PR and make a note it's related to  torch/cutorch#431.
- I am getting this error in cunn tests now, are you getting the same : 


SpatialDilatedConvolution_backward_single
error on state (backward)
LT failed: 2.8561930656433 >= 0.01
        /home/bfomitchev/git/R5/install/share/lua/5.1/cunn/test.lua:1738: in function 'v'
        /home/bfomitchev/git/R5/install/share/lua/5.1/cunn/test.lua:5576: in function </home/bfomitchev/git/R5/install/share/lua/5.1/cunn/test.lua:5574>